### DISCUSSION TITLE
Add import aliases

### DISCUSSION THREAD

**Discussion by spacey-sooty** [withastro/roadmap#719] [Proposal]
### Body

# Summary

Add import aliases similar to how they exist in NextJS to improve DX

# Background & Motivation

Coming from NextJS one of most annoying things when using Astro is importing a component or stylesheet from a really nested page. The constant "../../../components/name.astro" makes code less readable and worse to write overall.

# Goals

- improve DX in relation to imports from nested components
- improve readability in Astro imports

# Example

Old
```astro
---
import Name from '.../.../../../components/name.astro'
---
```

New
```astro
---
import Name from '@/components/name.astro'
---
```
Could also use symbols other than @ 

# Alternate Solution

Another solution I could see which to me seems harder to implement would be to use NextJS 13 style folder based routing instead of file based routing. This would allow you to put your components closer to where they are used instead of using a components folder.

**Comment by delucis** [comment]
Hi @spacey-sooty! This is already supported: https://docs.astro.build/en/guides/aliases/
